raspberry-uc-system/RaspberryUserControlSystem/RaspberryBackend/Components/I2C/Multiplexer.cs

```diff
@@ -43,6 +45,82 @@ public override void initiate()
         }
 
         /// <summary>
+        /// Sets the multiplexer configuation to a default HI from the XML Config file.
+        /// The HI is:
+        /// Family: "Pure", Model: "312 702 S (DN)"
+        /// </summary>
+        public void setMultiplexerConfiguration()
+        {
+            setMultiplexerConfiguration("Pure", "312 702 S (DN)");
+        }
+
+        /// <summary>
+        /// Sets the multiplexer configuation to a specific HI
+        /// </summary>
+        /// <param name="family">family name of the HI, e.g.: "Pure"</param>
+        /// <param name="model_name">model name of the HI: e.g: "312 702 S (DN)"</param>
+        public void setMultiplexerConfiguration(string family, string model_name)
+        {
+            Debug.WriteLine(this.GetType().Name + "::: Setting Multiplexer Config:");
+            MultiplexerConfig mux_config = MultiplexerConfigParser.getMultiplexerConfig(family, model_name);
+            Dictionary<int, string> X_to_value_map = mux_config.X_Pin_To_Value_Map;
+            Dictionary<string, int> value_to_Y_map = GPIOConfig._gpio_to_Y_map;
+
+            foreach (int value_x in X_to_value_map.Keys)
+            {
+                foreach (string y_value in value_to_Y_map.Keys)
+                {
```

Comment:
die logik kann man sicher noch intuitiver machen, aber ich war ja dabei und es junktioniert ja auch. Aus dem stehgreif fällt mir nichts konkretes dazu ein. 
Evtl kann man die zweite schleife rausmachen und die x_to_value_map Values ablaufen. ungefähr so: 
foreach( int xPin in X_to_value_map.Keys)
* string xValue = X_to_value_map[xPin];
* int yPin;
* wenn value_to_Y_map[xValue] nicht Null und nicht Leer 
** yPin = value_to_Y_map[xValue]
** connectPins(xPin, yPin)